src/InsertionsClient.Core/Api/Providers/InsertionApi.cs

```diff
@@ -225,10 +227,21 @@ internal bool TryExtractManifestAssets(string manifestFile, ICollection<Asset> a
 
                 Trace.WriteLine($"De-serialized {buildManifest.Builds.Count} builds from the manifest file.");
 
-                int initialCollectionSize = assets.Count;
+                if (!buildManifest.Builds.Any(b => b?.Assets != null && b.Assets.Any()))
+                {
+                    details = $"No assets were found in the manifest file.";
+                    return false;
+                }
 
                 foreach (Build build in buildManifest.Builds)
                 {
+                    if (buildFilter != null && !buildFilter(build))
+                    {
+                        Trace.WriteLine($"The build with {nameof(build.BuildNumber)} {build.BuildNumber} was ignored by the filter. " +
```

Comment:
Could `nameof(build.BuildNumber)` be cached out of the loop?  

if you CPU profiled this code, you may see a savings opportunity in removing this from the loop.

Reply:
`nameof` is evaluated at compile time, so this shouldn't be an issue.

Reply:
IL & Jit are the truth